Source/DafnyLanguageServer/Handlers/Custom/DafnyCounterExampleHandler.cs

```diff
@@ -30,9 +32,14 @@ public class DafnyCounterExampleHandler : ICounterExampleHandler {
       try {
         var projectManager = await projects.GetProjectManager(request.TextDocument);
         if (projectManager != null) {
-          await projectManager.VerifyEverythingAsync(request.TextDocument.Uri.ToUri());
+          var uri = request.TextDocument.Uri.ToUri();
+          await projectManager.VerifyEverythingAsync(uri);
 
-          var state = await projectManager.GetIdeStateAfterVerificationAsync();
+          var state = await projectManager.States.Select(s => s.Value).Where(s =>
+            s.Status == CompilationStatus.ResolutionSucceeded &&
```

Comment:
I feel like this could be written

```cs
s.EverythingFinished() // needs to be refactored.
```

what do you think?

Reply:
Not sure what you mean but I extracted some code into a method to make it more clear what it does.